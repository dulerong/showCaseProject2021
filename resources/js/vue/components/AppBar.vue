<template lang="pug">
v-app-bar(
  app
  flat
  height="64"
  outlined
)
  ToolTip(:message="$t('appBar.tooltip.sideDrawer')")
    v-app-bar-nav-icon(@click.stop="toggle" data-testid="toggleButton")
  v-spacer
  MenuLanguage
  ToolTip(:message="$t('appBar.otherSettings')")
    v-btn(icon @click.stop="toggleSide" data-testid="toggleSideButton")
      v-icon mdi-settings-outline
  v-divider(vertical class="mx-1" inset)
  AppBarProfileMenu
</template>

<script>
export default {
  components: {
    ToolTip: require('~components/ToolTip').default,
    MenuLanguage: require('~components/MenuLanguage').default,
    AppBarProfileMenu: require('~components/AppBarProfileMenu').default,
  },
  props: {
    toggle: {
      type: Function,
      required: true
    },
    toggleSide: {
      type: Function,
      required: true
    }
  },
}
</script>