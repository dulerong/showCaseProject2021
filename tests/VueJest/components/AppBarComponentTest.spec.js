import { mount, createLocalVue } from '@vue/test-utils'
import Vuetify from 'vuetify'

import AppBar from '~components/AppBar'

describe('AppBar Component', () => {
  const localVue = createLocalVue()

  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  afterEach(() => {    
    jest.clearAllMocks()
  })

  const mountFunction = options => {
    return mount(AppBar, {
      localVue,
      vuetify,
      mocks: {
        $t: (msg) => msg,
        $route: {
          name: 'home'
        }
      },
      stubs: {
        MenuLanguage: true,
        AppBarProfileMenu: true
      },
      propsData: {
        toggle: jest.fn(),
        toggleSide: jest.fn()
      },
      ...options
    })
  }

  it('Renders properly', () => {
    const wrapper = mountFunction()

    expect(wrapper.exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('AppBar Nav button: on click toggle fn is called', async () => {
    const wrapper = mountFunction()

    const button = wrapper.find('[data-testid="toggleButton"]')
    expect(button.exists()).toBe(true)

    await button.trigger('click')
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.toggle).toHaveBeenCalled()
  })

  it('Translate button: on click toggleSide fn is called', async () => {
    const wrapper = mountFunction()

    const button = wrapper.find('[data-testid="toggleSideButton"]')
    expect(button.exists()).toBe(true)

    await button.trigger('click')
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.toggleSide).toHaveBeenCalled()
  })
})