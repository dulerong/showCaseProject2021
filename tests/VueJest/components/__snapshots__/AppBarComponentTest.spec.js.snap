// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar Component Page renders properly 1`] = `
<header class="v-sheet theme--light v-toolbar v-app-bar v-app-bar--fixed" style="height: 64px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;">
  <div class="v-toolbar__content" style="height: 64px;"><span class="v-tooltip v-tooltip--bottom"><!----><div aria-haspopup="true" aria-expanded="false"><button type="button" class="v-app-bar__nav-icon v-btn v-btn--icon v-btn--round theme--light v-size--default" data-testid="toggleButton"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu theme--light"></i></span></button></div></span>
  <div class="spacer"></div>
  <div class="v-menu"><button type="button" class="pa-0 v-btn v-btn--text theme--light v-size--default" role="button" aria-haspopup="true" aria-expanded="false"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-translate theme--light"></i><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-down theme--light" style="font-size: 16px;"></i></span></button>
    <!---->
  </div><span class="v-tooltip v-tooltip--bottom"><!----><div aria-haspopup="true" aria-expanded="false"><button type="button" class="v-btn v-btn--icon v-btn--round theme--light v-size--default" data-testid="toggleSideButton"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-settings-outline theme--light"></i></span></button></div></span>
  <hr role="separator" aria-orientation="vertical" class="mx-1 v-divider v-divider--inset v-divider--vertical theme--light">
  <div class="v-menu"><button type="button" class="v-btn v-btn--has-bg theme--light v-size--default" style="height: 100%;" role="button" aria-haspopup="true" aria-expanded="false"><span class="v-btn__content"><div class="v-toolbar__title">Username</div></span></button>
    <!---->
  </div>
  </div>
</header>
`;
